File: DART/patterns/lib/patterns.dart
// A pattern matches a value, destructures it, both depending on the context and
// shape of the pattern.

//  MATCHING
const a = 'a';
const b = 'b';

String switchCase(String letter) {
  switch (letter) {
    // matches if letter is 'a'
    case a:
      return 'a';
    case b:
      return 'b';
    default:
      return 'not a or b';
  }
}

// pattern variable declaration
Object pattern_variable_declaration() {
  var (d, e) = ('str', [1, 2, 3]);
  return {d, e};
}

String pattern_destructure() {
  var (a, b) = ('left', 'right');
  // swap
  (b, a) = (a, b);
  return '$a, $b';
}

File: PYTHON/tutorials/patterns.py
def pattern(n):
    for i in range(0, n):
        for j in range(0, i+1):
            print("* ", end="")
        print("\r")


pattern(5)

print()


def pattern(n):
    k = 2 * n - 2
    for i in range(0, n):
        for j in range(0, k):
            print(end=" ")
        k = k - 2
        for j in range(0, i + 1):
            print("* ", end="")
        print("\r")


pattern(5)

print()


def pattern(n):
    for i in range(n, -1, -1):
        for j in range(0, i + 1):
            print("* ", end="")
        print("\r")


pattern(5)

print()
for i in range(5):
    for j in range(5):
        if i + j == 2 or i - j == 2 or i + j == 6 or j - i == 2:
            print("*", end="")
        else:
            print(end=" ")
    print()
